File: RuntimeWidget/app/src/main/java/com/example/runtimewidget/MainActivity.java
package com.example.runtimewidget;

import androidx.appcompat.app.AppCompatActivity;
import androidx.constraintlayout.widget.ConstraintLayout;

import android.os.Bundle;
import android.widget.DatePicker;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        //setContentView(R.layout.activity_main);
        ConstraintLayout layout = new ConstraintLayout(this);
        DatePicker datePicker = new DatePicker(this);
        layout.addView(datePicker);
        setContentView(layout);
    }
}

File: TransitionActivity/app/src/main/java/com/example/transitionactivity/MainActivity.kt
package com.example.transitionactivity

import android.app.ActivityOptions
import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.util.Pair as UtilPair
class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }

fun onClick(view: View) {

    val transitionImg = findViewById<View>(R.id.imgTaj)
    val transitionText = findViewById<View>(R.id.txvTaj)

    val imgPair = UtilPair.create(transitionImg, "sharedImage")
    val textPair = UtilPair.create(transitionText, "sharedTitle")

    val options = ActivityOptions.makeSceneTransitionAnimation(this, imgPair, textPair)
    val i = Intent(this@MainActivity, DetailActivity::class.java)
    startActivity(i, options.toBundle())

}}
